py-utils/src/utils/setup/consul/consul.py

```
@@ -0,0 +1,227 @@
+                        server=str(server).lower(),
+                        bootstrap_expect=2).dump(f"{config_path}/consul.hcl")
+        command = f"consul validate {config_path}/consul.hcl"
+        _, err, returncode = SimpleProcess(command).run()
+        if returncode != 0:
+            ConsulSetupError(returncode, 
+                            "Consul Setup config file %s validation failed with error :%s"
+                            %(f"{config_path}/consul.hcl", err))
+        command = f"chown -R consul:consul {config_path} {data_path}"
+        _, err, returncode = SimpleProcess(command).run()
+        if returncode != 0:
+            ConsulSetupError(returncode, 
+                            "Consul Setup changing ownership failed for %s %s with error: %s"
+                            %(config_path, data_path, err))
+        return 0
+
+    def run_as_server(self, server_node):
+        storage_set_count = int(Conf.get(self.index, f"cluster>{self.cluster_id}>site>storage_set_count"))
+
+        node_id = int(Conf.get(self.index, f"server_node>{server_node}>node_id"))
+        if storage_set_count < 5:
+            if node_id < 4:
+                return True
+        elif storage_set_count == 6:
+            if node_id < 6:
+                return True
+        elif storage_set_count == 12:
+            if node_id < 8:
+                return True
+        return False
+
+
+    def test(self, plan):
+        """ Perform configuration testing. Raises exception on error """
+
+        if "TestConsul" in globals():
+            test_consul = TestConsul()
+            test_consul.put()
+            test_consul.get()
+            test_consul.delete()
+        else:
+            print("Install cortx-py-utils-test to run test")
+        return 0
+
+    def reset(self):
+        """ Performs Configuraiton reset. Raises exception on error """
+
+        command = f"consul kv delete --recurse"
+        SimpleProcess(command).run()
```

Of the command fails, will `SimpleProcess.run`  raise an exception?
If not then, the error can be potentially swallowed.